bot/cogs/search.py

```diff
@@ -81,18 +80,16 @@ async def _basic_search(self, ctx, query: str, category: str) -> None:
                 results = await self._search_logic(query, is_nsfw, category)
             except SafesearchFail:
                 await ctx.send(f":x: Sorry {ctx.author.mention}, your message contains filtered words, I've removed this message.")
-                await ctx.message.delete()
-                return
+                return await ctx.message.delete()
```

Comment:
The return type of this function is `None` doing this would change that to `t.Optional[Message]`.
Since we enforce proper typing hints on all functions it would probably be better to keep this as it was since that way the return type won't change.

**Note: This also applies for all such occurrences**